pkg/data/domain/app/validate.go

```diff
@@ -0,0 +1,418 @@
+package app
+
+import (
+	"context"
+	"io/ioutil"
+	"net/http"
+	"os/exec"
+	"path"
+
+	applicationv1alpha1 "github.com/giantswarm/apiextensions/v2/pkg/apis/application/v1alpha1"
+	"github.com/xeipuuv/gojsonschema"
+	"gopkg.in/yaml.v3"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/giantswarm/microerror"
+
+	runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	valuesSchemaAnnotationKey = "application.giantswarm.io/values-schema"
+)
+
+// Validate attempts to validate the values of a deployed app (or multiple apps
+// depending on how this command is invoked) against the schema for those values.
+func (s *Service) Validate(ctx context.Context, options ValidateOptions) (ValidationResults, error) {
+	var err error
+
+	namespace := options.Namespace
+
+	// If the namespace is empty, set it to "default".
+	if namespace == "" {
+		namespace = defaultNamespace
+	}
+
+	// BUT if we want all namespaces, set it to 'metav1.NamespaceAll', aka ""
+	// again so the client gets all namespaces.
+	if options.AllNamespaces {
+		namespace = metav1.NamespaceAll
+	}
```

Comment:
In the `get clusters` command, we now do this defaulting at the command level

https://github.com/giantswarm/kubectl-gs/blob/78d6a3cad6e3f531b00622773e62c6d7a02db119/cmd/get/clusters/runner.go#L77